### PR TITLE
Weave-Models Tutorial

### PR DESCRIPTION
**Pull request by NiWaRe**
## Description

Add tutorial for how to integrate W&B Weave with W&B Models. Demo how to use a fine-tuned Chat Model from the Registry, integrate it into a RAG Model on Weave, evaluate the new RAG Model using Weave, and save the RAG Model back to the Registry. Working on this with @knisar.

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/tutorials/weave_models_registry.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+```
+
+# 2. Make `ChatModel` based on Artifact
+We retrieve the fine-tuned chat model from the Registry and create a `weave.Model` out of it that we can directly plug in to the [RagModel](https://wandb.ai/wandb-smle/weave-cookboook-demo/weave/object-versions?filter=%7B%22objectName%22%3A%22RagModel%22%7D&peekPath=%2Fwandb-smle%2Fweave-cookboook-demo%2Fobjects%2FRagModel%2Fversions%2FcqRaGKcxutBWXyM0fCGTR1Yk2mISLsNari4wlGTwERo%3F%26) in the next step. It takes in the same parameters as the existing [ChatModel](https://wandb.ai/wandb-smle/weave-cookboook-demo/weave/object-versions?filter=%7B%22objectName%22%3A%22RagModel%22%7D&peekPath=%2Fwandb-smle%2Fweave-rag-experiments%2Fobjects%2FChatModelRag%2Fversions%2F2mhdPb667uoFlXStXtZ0MuYoxPaiAXj3KyLS1kYRi84%3F%26) just the `init` and `predict` change.
+
+```python
+# pip install unsloth
+# Also get the latest nightly Unsloth!
+# pip uninstall unsloth -y && pip install --upgrade --no-cache-dir "unsloth[colab-new] @ git+https://github.com/unslothai/unsloth.git"
+```
```
Comment (vanpelt) [inline docs/tutorials/weave_models_registry.md:55]: Seems like we can just get rid of this code block?

File: docs/tutorials/weave_models_registry.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+ And finally run the evaluation in async mode:
+
+ ```python
+ await new_chat_model.predict(
+     [{"role": "user", "content": "What is the capital of Germany?"}]
+ )
+ ```
+
+ # 3. Integrate new `ChatModel` version into `RagModel`
+Now we retrieve the [RagModel](https://wandb.ai/wandb-smle/weave-cookboook-demo/weave/object-versions?filter=%7B%22objectName%22%3A%22RagModel%22%7D&peekPath=%2Fwandb-smle%2Fweave-cookboook-demo%2Fobjects%2FRagModel%2Fversions%2FcqRaGKcxutBWXyM0fCGTR1Yk2mISLsNari4wlGTwERo%3F%26) from our existing Weave project and exchange the `ChatModel` to the new one. We don't need to change or re-create any of the other components (VDB, prompts, etc.)!
```
Comment (vanpelt) [inline docs/tutorials/weave_models_registry.md]: I didn't see the definition of RagModel anywhere earlier in the cookbook, only a UnslothLoRAChatModel.  In the Unsloth model the chat_model parameter is a string pointing to the weave ref.  In this RagModel it seems to be an instantiated object.  It also wasn't clear in the docs how I get "weave:///wandb-smle/weave-cookboook-demo/object/RagModel:cqRaGKcxutBWXyM0fCGTR1Yk2mISLsNari4wlGTwERo" maybe a screenshot of the use tab or we could also just not talk about this weave ref magic, as it stands it's a bit confusing.

File: docs/tutorials/weave_models_registry.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+```python
+# pip install litellm faiss-gpu
```
Comment (vanpelt) [inline docs/tutorials/weave_models_registry.md]: These pip install comments are a little wonky, let's just add a requirements.in file or a single cell that does all the installing.

File: docs/tutorials/weave_models_registry.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+).get()
+# MAGIC: exchange chat_model and publish new version (no need to worry about other RAG components)
+RagModel.chat_model = new_chat_model
+# first publish new version so that in prediction we reference new version
+PUB_REFERENCE = weave.publish(RagModel, "RagModel")
+await RagModel.predict("When was the first conference on climate change?")
+```
+
+# 4. Run new `weave.Evaluation` connecting to the existing models run
+Finally, we can evaluate our new `RagModel` on of the existing `weave.Evaluation`. To make the integration as easy as possible we include the following changes.
```
Comment (vanpelt) [inline docs/tutorials/weave_models_registry.md]: Typo

**Review by vanpelt — APPROVED** [review]
Looks great!

File: docs/tutorials/weave_models_registry.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+Find the public workspace for both W&B Models and W&B Weave [here](https://wandb.ai/wandb-smle/weave-cookboook-demo/weave/evaluations).
+
+<img src="/images/tutorials/weave_models_workflow.jpg"  alt="Weights & Biases" />
+
+We'll cover the following steps as part of the workflow:
+
+1. Instrument our RAG application code with W&B Weave
+2. Fine-tune an LLM (we've used Llama 3.2 but this can be replaced with any other LLM) and track it with W&B Models
+3. Log the fine-tuned model to the Registry
```
Comment (ngrayluna) [inline docs/tutorials/weave_models_registry.md]: Registry -> W&B Registry 

Brownie points if we link to the W&B Registry docs the first time we mention it.

**Comment by johndmulhausen** [comment]
@NiWaRe A lot of good feedback from Vale that you should address, but love that this is happening.

File: docs/tutorials/weave_models_registry.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+```bash
+pip install unsloth
+pip uninstall unsloth -y && pip install --upgrade --no-cache-dir "unsloth[colab-new] @ git+https://github.com/unslothai/unsloth.git"
+```
+
+The model team fine-tuned different Llama-3.2 models using the `unsloth` library to make it faster. Hence we'll use the special `unsloth.FastLanguageModel` or `peft.AutoPeftModelForCausalLM` models with adapters to load in the model once we download it from the Registry. The loading code in the `model_post_init` can be simply copy & pasted from the "Use" tab in the Registry.
```
Comment (ngrayluna) [inline docs/tutorials/weave_models_registry.md]: Remove "simply". Replace "&" with "and".

**Comment by ngrayluna** [comment]
Heads up that in the notebook there is a floating `.` in 

```
os.environ["WANDB_API_KEY"] = userdata.get('WANDB_API_KEY').
```

**Review by ngrayluna — APPROVED** [review]

**Pull request merged**